src/charm_state.py

```diff
@@ -413,9 +414,21 @@ def from_charm(  # pylint: disable=too-many-arguments,too-many-positional-argume
             CharmConfigInvalidError: if the charm configuration is invalid.
         """
         try:
+            # Compute public_baseurl if not configured
+            config = dict(charm.config.items())
```

Comment:
```suggestion
            config = dict(charm.config)
```

The `dict` function automatically accepts mappings, so it doesn't require `.items()`. https://docs.python.org/3/library/stdtypes.html#dict

Reply:
thanks, I'll fix this in a follow-up PR